src/extras/template.py

```diff
@@ -27,7 +27,20 @@
 from inspect import getargspec
 
 from django.template import Context
-from django.template.base import generic_tag_compiler, TagHelperNode, Template
+try:
+    from django.template.base import generic_tag_compiler, TagHelperNode, Template
+except ImportError:
```

Comment:
It seems like this is a fine solution for now, but I think going forward unfortunately we're going to need a version check, and different code for each version with significant changes.  In particular, backporting any changes when we're already trying to merge different versions of the code seems like it's just an impossible mess.

Also, I think it's probably better to do an explicit version check instead of `try`/`except`?